File: ui/src/schema/sync.ts
import { z } from 'zod'
import { createApiResponseSchema } from '@/schema/api-response.ts'

export const SyncInterfacePreviewSchema = z.object({
  id: z.string(),
  disabled: z.boolean(),
  comment: z.string().nullable(),
  name: z.string(),
  listen_port: z.string(),
  mtu: z.string(),
  is_running: z.boolean(),
})

export const SyncPeerPreviewSchema = z.object({
  id: z.string(),
  disabled: z.boolean(),
  comment: z.string().nullable(),
  name: z.string(),
  interface: z.string(),
  allowed_address: z.string(),
})

export const SyncInterfacesResponseSchema = createApiResponseSchema(
  z.array(SyncInterfacePreviewSchema)
)

export const SyncPeersResponseSchema = createApiResponseSchema(
  z.array(SyncPeerPreviewSchema)
)

export const SyncInterfacesRequestSchema = z.object({
  interface_ids: z.array(z.string()),
})

export const SyncPeersRequestSchema = z.object({
  peer_ids: z.array(z.string()),
})

export type SyncInterfacePreview = z.infer<typeof SyncInterfacePreviewSchema>
export type SyncPeerPreview = z.infer<typeof SyncPeerPreviewSchema>
export type SyncInterfacesRequest = z.infer<typeof SyncInterfacesRequestSchema>
export type SyncPeersRequest = z.infer<typeof SyncPeersRequestSchema>

File: api/http/schema/sync.go
package schema

type SyncInterfacePreviewResponse struct {
	ID         string  `json:"id"`
	Disabled   bool    `json:"disabled"`
	Comment    *string `json:"comment"`
	Name       string  `json:"name"`
	ListenPort string  `json:"listen_port"`
	MTU        string  `json:"mtu"`
	IsRunning  bool    `json:"is_running"`
}

type SyncPeerPreviewResponse struct {
	ID             string  `json:"id"`
	Disabled       bool    `json:"disabled"`
	Comment        *string `json:"comment"`
	Name           string  `json:"name"`
	Interface      string  `json:"interface"`
	AllowedAddress string  `json:"allowed_address"`
}

type SyncInterfacesRequest struct {
	InterfaceIDs []string `json:"interface_ids"`
}

type SyncPeersRequest struct {
	PeerIDs []string `json:"peer_ids"`
}
